File: src/hunt_optimizer/model/SimpleQuestModel.ts
import { Episode } from "../../core/data_formats/parsing/quest/Episode";
import { NpcType } from "../../core/data_formats/parsing/quest/npc_types";

export class SimpleQuestModel {
    constructor(
        readonly id: number,
        readonly name: string,
        readonly episode: Episode,
        readonly enemy_counts: Map<NpcType, number>,
    ) {
        if (!id) throw new Error("id is required.");
        if (!name) throw new Error("name is required.");
        if (!enemy_counts) throw new Error("enemyCounts is required.");
    }
}

File: web/src/main/kotlin/world/phantasmal/web/huntOptimizer/models/SimpleQuestModel.kt
package world.phantasmal.web.huntOptimizer.models

import world.phantasmal.psolib.Episode
import world.phantasmal.psolib.fileFormats.quest.NpcType

class SimpleQuestModel(
    val id: Int,
    val name: String,
    val episode: Episode,
    val enemyCounts: Map<NpcType, Int>,
)
